### PR TITLE
Nodify, add Typings and bundle

### PR DESCRIPTION
**Pull request by alshakero**
Related to https://github.com/Palindrom/JSON-Patch-OT/issues/5.

This is 100% backward compatible. Yet 100% Node and future Palindrom compatible.

Tested with Browser `<script />`, Node CommonJS, Babel ES6 imports and TS imports. All Jasmine tests pass on Node and browser.

I took action earlier to save time. Any modifications and ideas are welcome.

After puplishing **`JSON-Patch-Queue`** on NPM. `package.json` needs to be updated to use NPM rather than Github.

### CODE REVIEW

**Comment by alshakero** [comment]
@tomalec ready for review.

File: test/SpecRunner.html [hunk 1]
```diff
@@ -22,7 +22,7 @@
   <!-- include source files here... -->
 
   <!-- include spec files here... -->
-  <script src="../../JSON-Patch-queue/src/json-patch-queue.js"></script>
+  <script src="https://rawgit.com/Palindrom/JSON-Patch-Queue/master/src/json-patch-queue.js"></script>
```
Comment (tomalec) [inline test/SpecRunner.html]: I'd prefer to be able to run it offline.
Can't we load it from `node_modules`?

File: test/SpecRunnerMinified.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  <script src="lib/benchmark.js/benchmark.js"></script>
+  <script src="lib/benchmark_reporter.js"></script>
+  <link rel="stylesheet" type="text/css" href="lib/benchmark_reporter.css">
+
+  <!-- include source files here... -->
+
+  <!-- include spec files here... -->
+  <!-- needless here <script src="https://rawgit.com/Palindrom/JSON-Patch-Queue/master/src/json-patch-queue.js"></script> -->
+  
+  <script src="../dist/json-patch-ot-agent.bundle.js"></script>
```
Comment (tomalec) [inline test/SpecRunnerMinified.html]: shouldn't we load `.min` for `..Minified.html`?

File: test/SpecRunnerMinified.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  <!-- include Benchmark.js -->
+  <script src="lib/benchmark.js/vendor/lodash/dist/lodash.compat.js"></script>
+  <script src="lib/benchmark.js/vendor/platform.js/platform.js"></script>
+  <script src="lib/benchmark.js/benchmark.js"></script>
+  <script src="lib/benchmark_reporter.js"></script>
+  <link rel="stylesheet" type="text/css" href="lib/benchmark_reporter.css">
+
+  <!-- include source files here... -->
+
+  <!-- include spec files here... -->
```
Comment (tomalec) [inline test/SpecRunnerMinified.html]: Those comments does not match.

File: test/SpecRunnerMinified.html (hunk 2 above)
Comment (alshakero) [inline test/SpecRunnerMinified.html]: Doesn't matter, I'm testing the bundling, i.e.: without
```html
<script src="https://rawgit.com/Palindrom/JSON-Patch-Queue/master/src/json-patch-queue.js"></script>`
```

File: bower.json [hunk 4]
```diff
@@ -20,8 +20,7 @@
     "node_modules",
     "bower_components",
     "test",
-    "tests",
-    "gruntfile.js"
+    "tests"
```
Comment (tomalec) [inline bower.json]: I'd still keep it ignored

File: src/json-patch-ot-agent.js [hunk 5]
```diff
@@ -1,15 +1,21 @@
...
+		var JSONPatchQueue = require('json-patch-queue').JSONPatchQueue;
+	}
+	else {
+		throw new Error('You need to reference JSONPatchQueue before JSONPatchOTAgent');
+	}
 }
+
 /**
  * [JSONPatchOTAgent description]
+ * version: 1.1.0
```
Comment (tomalec) [inline src/json-patch-ot-agent.js]: why not `@version`?

Comment (alshakero) [inline src/json-patch-ot-agent.js]: Grunt doesn't detect it and I'm too exhausted to investigate.

File: bower.json (hunk 4 above)
Comment (alshakero) [inline bower.json]: Yes, this is a mistake, because I removed it initially. Nice catch.

File: src/json-patch-ot-agent.js (hunk 5 above)
Comment (tomalec) [inline src/json-patch-ot-agent.js]: Previously it was detecting everything that matches `/version\s*:/`.
In one of our repos we user custom regex to detect that in other files as well.

Comment (alshakero) [inline src/json-patch-ot-agent.js]: Then I guess it's the ":".

Comment (alshakero) [inline src/json-patch-ot-agent.js]: Worked.

**Comment by alshakero** [comment]
@tomalec @warpech just to make sure and/or clarify, this one depends on `json-patch-queue`, so a bundle is actually needed, does this comment (https://github.com/Palindrom/JSON-Patch-OT/issues/5#issuecomment-285670232) apply to it, or I should keep bundles?

**Comment by warpech** [comment]
I think bundle is not needed, because both `JSON-Patch-OT-agent` and `json-patch-queue` are only meant to be used the NPM way.

**Pull request merged**